src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -474,7 +470,7 @@ private boolean hasMultipleLinesBefore(DetailAST ast) {
      */
     private void processPackage(DetailAST ast, DetailAST nextToken) {
         if (ast.getLineNo() > 1 && !hasEmptyLineBefore(ast)) {
-            if (getFileContents().getFileName().endsWith("package-info.java")) {
+            if (getFileContents().inPackageInfo()) {
```

Comment:
would be awesome to move to separate commit (first commit of this PR), as it is not related issue

Reply:
done